charms/istio-pilot/src/virtual_service.yaml.j2

```diff
@@ -6,6 +6,9 @@ metadata:
 spec:
   gateways:
   - '{{ namespace }}/{{ gateway }}'
+  {% for i in custom_gateway %}
```

Comment:
```suggestion
  {% for i in custom_gateways %}
```

Reply:
Could we do `gateways = [gateway] + custom_gateways` in the charm code, then include them in the template as `{{ gateways | tojson | safe }}`?

Reply:
Yes, sure!

Reply:
After trying to the aggregated way, I still think the original is much simpler.
The aggregated one needs to handle the logic of `{{ namespace }}/{{ gateway }}`
It makes the code inside the charm code really complicated because the namespace is extracted inside a loop.
Therefore, I prefer to keep the current approach so that it looks simpler (but of course looks dumb too)